adapters/synacormedia/synacormedia.go

```diff
@@ -0,0 +1,184 @@
+package synacormedia
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/golang/glog"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type SynacorMediaAdapter struct {
+	EndpointTemplate template.Template
+}
+
+type SyncEndpointTemplateParams struct {
+	SeatId string
+}
+
+type ReqExt struct {
+	SeatId string `json:"seatId"`
+}
+
+func NewSynacorMediaBidder(endpointTemplate string) adapters.Bidder {
+	syncTemplate, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		glog.Fatal("Unable to parse endpoint url template")
+		glog.Fatal(endpointTemplate)
+		return nil
+	}
+	return &SynacorMediaAdapter{EndpointTemplate: *syncTemplate}
+}
+
+func (a *SynacorMediaAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var bidRequests []*adapters.RequestData
+
+	adapterReq, errors := a.makeRequest(request)
+	if adapterReq != nil {
+		bidRequests = append(bidRequests, adapterReq)
+	}
+	errs = append(errs, errors...)
+
+	return bidRequests, errs
+}
+
+func (a *SynacorMediaAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb.Imp
+	var re *ReqExt
+
+	for _, imp := range request.Imp {
+		_, err := getExtImpObj(&imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		validImps = append(validImps, imp)
+	}
+
+	if len(validImps) == 0 {
+		return nil, errs
+	}
+
+	var err error
+	// need to grab an impression to get the seat id
+	var firstImp = validImps[0]
+	firstExtImp, err := getExtImpObj(&firstImp)
+	if err != nil {
+		return nil, append(errs, err)
+	}
```

Comment:
This is a minor performance thing but, we can probably avoid the overhead of executing the `getExtImpObj(imp *openrtb.Imp)` twice on the first element of `validImps` with a minor workaround like: 
```
 52 func (a *SynacorMediaAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
 53     var errs []error
 54     var validImps []openrtb.Imp
 55     var re *ReqExt
     +  var firstExtImp *openrtb_ext.ExtImpSynacormedia = nil
 56
 57     for _, imp := range request.Imp {
 58  -      _, err := getExtImpObj(&imp)
     +      validImp, err := getExtImpObj(&imp)
 59         if err != nil {
 60             errs = append(errs, err)
 61             continue
 62         }
 63         validImps = append(validImps, imp)
     +      if firstExtImp == nil {
     +          firstExtImp = validImp
     +      }
 64     }
 65
 66     if len(validImps) == 0 {
 67         return nil, errs
 68     }
 69
 70     var err error
 71     // need to grab an impression to get the seat id
 72  -  var firstImp = validImps[0]
 73  -  firstExtImp, err := getExtImpObj(&firstImp)
 74  -  if err != nil {
 75  -      return nil, append(errs, err)
 76  -  }
 77
 78     if firstExtImp.SeatId == "" {
 79         return nil, append(errs, &errortypes.BadServerResponse{
 80             Message: fmt.Sprintf("Impression missing seat id"),
 81         })
 82     }
 83     re = &ReqExt{SeatId: firstExtImp.SeatId}
 84
 85 +-- 21 lines: create JSON Request Body--------------------------------------------------------------------
106
107     return &adapters.RequestData{
108         Method:  http.MethodPost,
109         Uri:     reqUri,
110         Body:    reqJSON,
111         Headers: headers,
112     }, errs
113 }
adapters/synacormedia/synacormedia.go
```

Reply:
Looks pretty good